tests/php/Extension/Engine/SiteTreePublishingEngineTest.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+namespace SilverStripe\StaticPublishQueue\Test\Extension\Engine;
+
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\Control\Director;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\StaticPublishQueue\Extension\Engine\SiteTreePublishingEngine;
+use SilverStripe\StaticPublishQueue\Extension\Publishable\PublishableSiteTree;
+use SilverStripe\StaticPublishQueue\Job\DeleteStaticCacheJob;
+use SilverStripe\StaticPublishQueue\Job\GenerateStaticCacheJob;
+use SilverStripe\StaticPublishQueue\Test\QueuedJobsTestService;
+use Symbiote\QueuedJobs\Services\QueuedJob;
+use Symbiote\QueuedJobs\Services\QueuedJobHandler;
+use Symbiote\QueuedJobs\Services\QueuedJobService;
+use Symbiote\QueuedJobs\Tests\QueuedJobsTest\QueuedJobsTest_Handler;
+
+class SiteTreePublishingEngineTest extends SapphireTest
+{
+    protected static $fixture_file = 'SiteTreePublishingEngineTest.yml';
+
+    protected static $required_extensions = [
+        SiteTree::class => [
+            PublishableSiteTree::class,
+            SiteTreePublishingEngine::class,
+        ],
+    ];
+
+    public function testPublishRecursive(): void
+    {
+        // Inclusion of parent/child is tested in PublishableSiteTreeTest
+        SiteTree::config()->set('regenerate_parents', PublishableSiteTree::RELATION_INCLUDE_NONE);
+        SiteTree::config()->set('regenerate_children', PublishableSiteTree::RELATION_INCLUDE_NONE);
+
+        /** @var QueuedJobsTestService $service */
+        $service = Injector::inst()->get(QueuedJobService::class);
```

Comment:
```suggestion
        $service = QueuedJobService::singleton();
```
[nitpick] Same thing but comes with return type hint and less code. Applies to multiple instances of this in this change-set.